### ISSUE TITLE
Problem with 33.1.1.633 on iOS 9

### ISSUE THREAD

**Issue by Bilbosax** [airsdk/Adobe-Runtime-Support#1275]
This morning, I upgraded to AIR 33.1.1.633 and tried to install my app on an iPad running iOS 9.3.6.  I got the error message displayed below.  If I switched back to 33.1.1.406, I had no problems.  So it must be a problem involving this SDK.  Any ideas what this means???

**WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/DealEstates.app/SC_Info/DealEstates.sinf in archive!
ERROR: Install failed. Got error "IncorrectArchitecture" with code 0x00000000: Failed to find matching arch for 64-bit Mach-O input file /private/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.rI7k2H/extracted/Payload/DealEstates.app/DealEstates**

**Comment by Bilbosax** [comment]
@FliplineStudios Seriously?!?  I thought that AIR worked on both 32 and 64-bit platforms and packaged both.  So you are saying I can't use this 32-bit iPad as a testing platform anymore???

**Comment by FliplineStudios** [comment]
Yup, in the past it's always worked for both 32-bit and 64-bit platforms, but starting with 33.1.1.620 Harman dropped support for 32-bit devices, I believe because they're generating the stub files for the SDK on newer machines now instead of the old ones that still supported 32-bit.

It's a bit disappointing since a good chunk of our app sales have been from older 32-bit iPads over the years, though at least for recent releases I haven't seen very many installs on them anymore.  We do use the older iPads for playtesting ourselves, since the performance of a 10-year-old iPad is sadly pretty close to the performance of cheapo discount Android tablets (for gpu vector graphics at least) so it was a nice benchmark for testing.  Guess we need to hunt down the oldest 64-bit iPad we can find for future testing now!

**Comment by marchbold** [comment]
My understanding is they can get the last version available for their device if they had previously installed it, I don't believe you can get new users on devices with old ios versions, but I'm not 100% sure on that one.

**Comment by spielepalast** [comment]
We have the same issue: our app updates are not installable anymore on IOS version 9. What can we do about it?

**Comment by ajwfrost** [comment]
@spielepalast is this on any iOS 9 device, or only on 32-bit ones?

There are two potential things at play here:
a) we don't have 32-bit support any more, due to the way in which the AOT compiler works (it needs to run with the same address size as the target, so a 32-bit process to create a 32-bit iOS binary, and we can't run 32-bit processes with the latest Xcode now)
b) there's a flag in the binaries which specifies a minimum OS level: we have this set to 9 by default but it can be raised by the application descriptor or by any ANE platform settings. So worth seeing if that has pushed the requirement up, which would stop it from running even on 64-bit iOS 9 devices.

As an aside .. we were looking at switching the avmplus code to always use the same 64-bit atomic type regardless of whether it's running on 32-bit or 64-bit machines. This would make the memory usage a lot more on 32-bit computers but they're fairly rare nowadays (apart from the fact that the default AIR builds on Windows are still 32-bit...!). If that makes a difference to this limitation of generating AOT code for a 32-bit binary, we may be able to re-introduce support...

thanks
